core/notifications/src/in_app_notification/repo.rs

```diff
@@ -9,6 +9,13 @@ pub(super) struct InAppNotificationsRepo {
     pool: PgPool,
 }
 
+pub struct NewInAppNotificationData {
```

Comment:
Lets go with the builder pattern that we are using extensively in Bria - no need to re-invent things we have already got a standard solution for.